tests/src/Context/JoinupCommunityContentContext.php

```diff
@@ -260,4 +263,124 @@ public function pinCommunityContentInGroup(AfterNodeCreateScope $scope) {
     }
   }
 
+  /**
+   * Asserts that a sequence of HTML markup exists inside a comment.
+   *
+   * The comment is identified by its '1 based' number. Numbers are starting
+   * from 1 and are assigned in the order comments are shown on the page,
+   * regardless of their indent.
+   *
+   * @param string $comment_number
+   *   The '1 based' number of the comment in the page.
+   * @param string $markup
+   *   The piece of markup to be checked.
+   *
+   * @Then (the )comment #:comment_number should contain the markup :markup
+   */
+  public function assertCommentContainsMarkup(string $comment_number, string $markup): void {
+    $comment = $this->getNumberedComment($comment_number - 1);
+    $regex = '/' . preg_quote($markup, '/') . '/ui';
+    Assert::assertRegExp($regex, $comment->getOuterHtml());
+  }
+
+  /**
+   * Asserts that a comment has given indent.
+   *
+   * The comment is identified by its '1 based' number. Numbers are starting
+   * from 1 and are assigned in the order comments are shown on the page,
+   * regardless of their indent. The indent is a '0 based' integer.
+   *
+   * @param string $comment_number
+   *   The '1 based' number of the comment in the page.
+   * @param string $indent
+   *   The '0 based' indent value.
```

Comment:
Very good idea to make this very clear in the documentation about the distinction between the 1-based comment numbers and 0-based indentation values. It could get confusing otherwise!